src/helpers/queryHelper.js

```diff
@@ -0,0 +1,166 @@
+      where: { status: search },
+      include: [
+        { model: models.User, where: { id: user }, as: 'requester' },
+        { model: models.Accommodation },
+      ],
+    });
+    return results;
+  }
+
+  if (userRole === 3) {
+    const results = await models.TripRequest.findAll({
+      where: { status: search },
+      include: [
+        { model: models.User, where: { id: user }, as: 'requester' },
+        { model: models.User, where: { id: user }, as: 'manager' },
+        { model: models.Accommodation },
+      ],
+    });
+    return results;
+  }
+};
+
+const byDepartureAndDestination = async (
+  origin,
+  destination,
+  user,
+  userRole
+) => {
+  const searchOne = origin.toLowerCase();
+  const searchTwo = destination.toLowerCase();
+  if (userRole === 2) {
+    const results = await models.TripRequest.findAll({
+      where: { departure_place: searchOne, destination: searchTwo },
+      include: [
+        { model: models.User, where: { id: user }, as: 'manager' },
+        {
+          model: models.Accommodation,
+        },
+      ],
+    });
+    return results;
+  }
+  if (userRole === 1) {
+    const results = await models.TripRequest.findAll({
+      where: { departure_place: searchOne, destination: searchTwo },
+      include: [
+        { model: models.User, where: { id: user }, as: 'requester' },
+        {
+          model: models.Accommodation,
+        },
+      ],
+    });
+    return results;
+  }
+
+  if (userRole === 3) {
+    const results = await models.TripRequest.findAll({
+      where: { departure_place: searchOne, destination: searchTwo },
+      include: [
+        { model: models.User, where: { id: user }, as: 'requester' },
+        { model: models.User, where: { id: user }, as: 'manager' },
+        {
+          model: models.Accommodation,
+        },
+      ],
+    });
+    return results;
+  }
+};
+
+const byDestination = async (destination, userRole, user) => {
+  const search = destination.toLowerCase();
+  if (userRole === 3) {
+    const results = await models.TripRequest.findAll({
+      where: { destination: search },
+      include: [
+        { model: models.User, as: 'manager' },
+        { model: models.User, as: 'requester' },
+        { model: models.Accommodation },
+      ],
+    });
+    return results;
+  }
+  if (userRole === 2) {
+    const results = await models.TripRequest.findAll({
+      where: { destination: search },
+      include: [
+        { model: models.User, as: 'manager' },
+        { model: models.Accommodation },
+      ],
+    });
+    return results;
+  }
+  if (userRole === 1) {
+    const results = await models.TripRequest.findAll({
+      where: { destination: search },
+      include: [
+        { model: models.User, as: 'requester' },
+        { model: models.Accommodation },
+      ],
+    });
+    return results;
+  }
+};
+
+export { byDeparture, byStatus, byDepartureAndDestination, byDestination };
```

Comment:
This file has a lot of code repetition in the condition role checks. Make that code re useable